### PR TITLE
JAVA-2528: Store suppressed exceptions in AllNodesFailedException

### PR DESCRIPTION
**Pull request by adutra**

### CODE REVIEW

**Review by tomekl007 — APPROVED** [review]

File: core/src/main/java/com/datastax/oss/driver/api/core/AllNodesFailedException.java [hunk 1]
```diff
@@ -31,7 +31,8 @@
 
 /**
  * Thrown when a query failed on all the coordinators it was tried on. This exception may wrap
- * multiple errors, use {@link #getAllErrors()} to inspect individual problems on each node.
+ * multiple {@linkplain #getSuppressed() suppressed exceptions}, use {@link #getAllErrors()} to
+ * inspect individual problems on each node.
```
Comment (olim7t) [inline core/src/main/java/com/datastax/oss/driver/api/core/AllNodesFailedException.java]: Suggestion: "This exception may wrap multiple errors, that are available either as suppressed exceptions, or via getAllErrors() where they are grouped by node".

**Review by olim7t — APPROVED** [review]
LGTM.

Overall I think it's a good change, but I'd just like to emphasize the fact that stack traces include suppressed exceptions. Here's an
[example](https://gist.github.com/olim7t/1a50f307939c51a13275ef8dcf35fff6) with bad credentials on a 3-node cluster.

One risk is that we get backlash because stack traces are getting too long. But I think with typical deployments (number of contact points, number of retries triggered by the policy), the number of child exceptions should remain low. Also, our client-side representations of server errors (`UnavailableException`, etc.) don't fill their stack trace, so they won't take too much space.

**Comment by tomekl007** [comment]
> LGTM.
> 
> Overall I think it's a good change, but I'd just like to emphasize the fact that stack traces include suppressed exceptions. Here's an
> [example](https://gist.github.com/olim7t/1a50f307939c51a13275ef8dcf35fff6) with bad credentials on a 3-node cluster.
> 
> One risk is that we get backlash because stack traces are getting too long. But I think with typical deployments (number of contact points, number of retries triggered by the policy), the number of child exceptions should remain low. Also, our client-side representations of server errors (`UnavailableException`, etc.) don't fill their stack trace, so they won't take too much space.

Maybe we could add some kind of a threshold parameter (similar to dsbulk) for setting the upper bound of number of suppressed exceptions?

**Comment by adutra** [comment]
> Maybe we could add some kind of a threshold parameter (similar to dsbulk) for setting the upper bound of number of suppressed exceptions?

Valid concern, but let's wait until someone complains before making this class any more complex.

**Pull request merged**